acpi/src/lib.rs

```diff
@@ -115,170 +138,150 @@ where
+                    unsafe { &*(self as *const Self as *const sdt::SdtHeader) }
+                }
             }
-            signature => {
-                self.sdts.insert(signature, Sdt { physical_address, length: header.length, validated: false });
-            }
-        }
 
-        Ok(())
+            let dsdt_address = fadt.dsdt_address()?;
+            let dsdt = unsafe { read_table::<H, Dsdt>(self.handler.clone(), dsdt_address)? };
+
+            Ok(AmlTable::new(dsdt_address, dsdt.header().length))
+        })
     }
 
-    /// Create a mapping to a SDT, given its signature. This validates the SDT if it has not already been
-    /// validated.
-    ///
-    /// ### Safety
-    /// The table's memory is naively interpreted as a `T`, and so you must be careful in providing a type that
-    /// correctly represents the table's structure. Regardless of the provided type's size, the region mapped will
-    /// be the size specified in the SDT's header. Providing a `T` that is larger than this, *may* lead to
-    /// page-faults, aliasing references, or derefencing uninitialized memory (the latter two of which are UB).
-    /// This isn't forbidden, however, because some tables rely on `T` being larger than a provided SDT in some
-    /// versions of ACPI (the [`ExtendedField`](crate::sdt::ExtendedField) type will be useful if you need to do
-    /// this. See our [`Fadt`](crate::fadt::Fadt) type for an example of this).
-    pub unsafe fn get_sdt<T>(&self, signature: sdt::Signature) -> Result<Option<PhysicalMapping<H, T>>, AcpiError>
-    where
-        T: AcpiTable,
-    {
-        let sdt = match self.sdts.get(&signature) {
-            Some(sdt) => sdt,
-            None => return Ok(None),
-        };
-        let mapping = unsafe { self.handler.map_physical_region::<T>(sdt.physical_address, sdt.length as usize) };
-
-        if !sdt.validated {
-            mapping.header().validate(signature)?;
+    /// Iterates through all of the SSDT tables.
+    pub fn ssdts(&self) -> SsdtIterator<H> {
+        let table_base_address = unsafe { self.mapping.virtual_start().as_ptr().add(1) as usize };
+        let table_end_address = table_base_address
+            + (((self.mapping.length as usize) - mem::size_of::<SdtHeader>()) / mem::size_of::<u64>());
+
+        SsdtIterator {
+            cur_address: table_base_address,
+            end_address: table_end_address,
+            handler: self.handler.clone(),
+            marker: core::marker::PhantomData,
         }
-
-        Ok(Some(mapping))
     }
 
     /// Convenience method for contructing a [`PlatformInfo`](crate::platform::PlatformInfo). This is one of the
     /// first things you should usually do with an `AcpiTables`, and allows to collect helpful information about
     /// the platform from the ACPI tables.
-    pub fn platform_info(&self) -> Result<PlatformInfo, AcpiError> {
-        PlatformInfo::new(self)
+    #[cfg(feature = "allocator_api")]
```

Comment:
Nice that we can keep this functionality. In a future PR, it'd be nice to see if we can feature-gate a `new` method back in with the default allocator

Reply:
Yeah, that's something I planned on doing in a separate PR, to try and keep this one's scope a little narrower.